### PR TITLE
ci: Use mergify for PR labels

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Changelog

- Build/Testing/CI

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/4009

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by Xuanwo** [comment]
For preview: https://github.com/Xuanwo/mergify-test/pull/1

File: .github/mergify.yml [hunk 1]
```diff
@@ -45,3 +43,71 @@ pull_request_rules:
...
+        add:
+          - pr-bugfix
+  - name: label on Improvement
+    conditions:
+      - body~=Improvement
+    actions:
+      label:
+        add:
+          - pr-improvement
+  - name: label on Improvement
```
Comment (BohuTANG) [inline .github/mergify.yml:79]: It seems this label will never be set since the above:
```
  conditions:
      - body~=Improvement
  ```

Comment (Xuanwo) [reply .github/mergify.yml:79]: They will be applied at the same time. Maybe we can adopt more strict rules for the behavior.

Comment (Xuanwo) [reply .github/mergify.yml:79]: image

Comment (Xuanwo) [reply .github/mergify.yml]: Fixed by checking `- Improvement` instead of `Improvement`

**Comment by Xuanwo** [comment]
Take https://github.com/Xuanwo/mergify-test/pull/2 for preview.

**Pull request merged**